subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/ecb.c

```diff
@@ -16,17 +17,82 @@
 
 #include "hal/debug.h"
 
+#if defined(NRF54L_SERIES)
+#define NRF_ECB                   NRF_ECB00
+#define ECB_IRQn                  ECB00_IRQn
+#define ECB_INTENSET_ERRORECB_Msk ECB_INTENSET_ERROR_Msk
+#define ECB_INTENSET_ENDECB_Msk   ECB_INTENSET_END_Msk
+#define TASKS_STARTECB            TASKS_START
+#define TASKS_STOPECB             TASKS_STOP
+#define EVENTS_ENDECB             EVENTS_END
+#define EVENTS_ERRORECB           EVENTS_ERROR
+#define NRF_ECB_TASK_STARTECB     NRF_ECB_TASK_START
+#define NRF_ECB_TASK_STOPECB      NRF_ECB_TASK_STOP
+#define ECBDATAPTR                IN.PTR
+
+struct ecb_job_ptr {
+	void *ptr;
+	struct {
+		uint32_t length:24;
+		uint32_t attribute:8;
+	} __packed;
+} __packed;
+#endif /* NRF54L_SERIES */
+
 struct ecb_param {
 	uint8_t key[16];
 	uint8_t clear_text[16];
 	uint8_t cipher_text[16];
+
+#if defined(NRF54L_SERIES)
+	struct ecb_job_ptr in[2];
+	struct ecb_job_ptr out[2];
+#endif /* NRF54L_SERIES */
 } __packed;
 
-static void do_ecb(struct ecb_param *ecb)
+static void do_ecb(struct ecb_param *ep)
 {
 	do {
 		nrf_ecb_task_trigger(NRF_ECB, NRF_ECB_TASK_STOPECB);
-		NRF_ECB->ECBDATAPTR = (uint32_t)ecb;
+
+#if defined(NRF54L_SERIES)
+		NRF_ECB->KEY.VALUE[3] = ((uint32_t)ep->key[0] << 24) |
+					((uint32_t)ep->key[1] << 16) |
+					((uint32_t)ep->key[2] << 8) |
+					(uint32_t)ep->key[3];
+		NRF_ECB->KEY.VALUE[2] = ((uint32_t)ep->key[4] << 24) |
+					((uint32_t)ep->key[5] << 16) |
+					((uint32_t)ep->key[6] << 8) |
+					(uint32_t)ep->key[7];
+		NRF_ECB->KEY.VALUE[1] = ((uint32_t)ep->key[8] << 24) |
+					((uint32_t)ep->key[9] << 16) |
+					((uint32_t)ep->key[10] << 8) |
+					(uint32_t)ep->key[11];
+		NRF_ECB->KEY.VALUE[0] = ((uint32_t)ep->key[12] << 24) |
+					((uint32_t)ep->key[13] << 16) |
+					((uint32_t)ep->key[14] << 8) |
+					(uint32_t)ep->key[15];
```

Comment:
Suggestion: Maybe you want to put all this behind a macro or inline function and share it below? (in case you'd feel like optimizing this a bit later, so it would be easier to change this to be a plain copy and ifdef the reversing/not reversing in `ecb_encrypt[_be]`)